src/services/has-many-getter.js

```diff
@@ -1,134 +1,57 @@
-      params,
-      this.fieldNamesRequested,
-    );
-  }
-
-  async buildWhereConditions({ associationName, search, filters }) {
-    const { AND } = this.operators;
-    const where = { [AND]: [] };
-
-    if (search) {
-      const searchCondition = this.searchBuilder.perform(associationName);
-      where[AND].push(searchCondition);
-    }
-
-
-    if (filters) {
-      const formattedFilters = await this.filtersParser.perform(filters);
-      where[AND].push(formattedFilters);
-    }
+class HasManyGetter extends ResourcesGetter {
+  constructor(model, association, lianaOptions, params) {
+    super(association, lianaOptions, params);
 
-    return where;
+    this._rootModel = model;
   }
 
-  async findQuery(queryOptions) {
-    if (!queryOptions) { queryOptions = {}; }
-    const { associationName, recordId } = this.params;
+  async _getRecords() {
+    const { associationName, recordId } = this._params;
+    const options = await this._buildQueryOptions();
 
-    const where = await this.buildWhereConditions(this.params);
-    const include = this.queryBuilder.getIncludes(this.association, this.fieldNamesRequested);
+    const record = await this._rootModel.findOne({
+      // Don't fetch parent attributes (perf)
+      attributes: [],
 
-    const record = await orm.findRecord(this.model, recordId, {
-      order: queryOptions.order,
+      // We are ordering with the relation
+      // https://github.com/sequelize/sequelize/issues/4553#issuecomment-213989980
+      order: options.order.map((o) => [this._model, ...o]),
+      offset: options.offset,
+      limit: options.limit,
+      where: new CompositeKeysManager(this._rootModel).getRecordsConditions([recordId]),
       subQuery: false,
-      offset: queryOptions.offset,
-      limit: queryOptions.limit,
-      // NOTICE: by default, all fields from the parent model
-      //         are retrieved, which can cause performance issues,
-      //         whereas we are only requesting the child model here
-      //         and we don't need the parent's attributes
-      attributes: [],
       include: [{
-        model: this.association,
+        model: this._model,
         as: associationName,
         scope: false,
         required: false,
-        where,
-        include,
+        where: options.where,
+        include: options.include,
       }],
     });
 
     return (record && record[associationName]) || [];
   }
 
   async count() {
-    const { associationName, recordId } = this.params;
-    const where = await this.buildWhereConditions(this.params);
-    const include = this.queryBuilder.getIncludes(this.association, this.fieldNamesRequested);
+    const { associationName, recordId } = this._params;
+    const options = await this._buildQueryOptions(true);
 
-    return this.model.count({
-      where: { [this.primaryKeyModel]: recordId },
+    return this._rootModel.count({
+      where: new CompositeKeysManager(this._rootModel).getRecordsConditions([recordId]),
```

Comment:
It is a bug fix (that nobody asked me to do)

On the previous version, I don't see how it could work when exploring the "hasMany" page of a model which is using a composite primary key.